src/java/org/apache/cassandra/utils/ForwardingMBeanServer.java

```diff
@@ -0,0 +1,274 @@
+    {
+        delegate().addNotificationListener(name, listener, filter, handback);
+    }
+
+    @Override
+    public void addNotificationListener(ObjectName name, ObjectName listener, NotificationFilter filter, Object handback) throws InstanceNotFoundException
+    {
+        delegate().addNotificationListener(name, listener, filter, handback);
+    }
+
+    @Override
+    public void removeNotificationListener(ObjectName name, ObjectName listener) throws InstanceNotFoundException, ListenerNotFoundException
+    {
+        delegate().removeNotificationListener(name, listener);
+    }
+
+    @Override
+    public void removeNotificationListener(ObjectName name, ObjectName listener, NotificationFilter filter, Object handback) throws InstanceNotFoundException, ListenerNotFoundException
+    {
+        delegate().removeNotificationListener(name, listener, filter, handback);
+    }
+
+    @Override
+    public void removeNotificationListener(ObjectName name, NotificationListener listener) throws InstanceNotFoundException, ListenerNotFoundException
+    {
+        delegate().removeNotificationListener(name, listener);
+    }
+
+    @Override
+    public void removeNotificationListener(ObjectName name, NotificationListener listener, NotificationFilter filter, Object handback) throws InstanceNotFoundException, ListenerNotFoundException
+    {
+        delegate().removeNotificationListener(name, listener, filter, handback);
+    }
+
+    @Override
+    public MBeanInfo getMBeanInfo(ObjectName name) throws InstanceNotFoundException, IntrospectionException, ReflectionException
+    {
+        return delegate().getMBeanInfo(name);
+    }
+
+    @Override
+    public boolean isInstanceOf(ObjectName name, String className) throws InstanceNotFoundException
+    {
+        return delegate().isInstanceOf(name, className);
+    }
+
+    @Override
+    public Object instantiate(String className) throws ReflectionException, MBeanException
+    {
+        return delegate().instantiate(className);
+    }
+
+    @Override
+    public Object instantiate(String className, ObjectName loaderName) throws ReflectionException, MBeanException, InstanceNotFoundException
+    {
+        return delegate().instantiate(className, loaderName);
+    }
+
+    @Override
+    public Object instantiate(String className, Object[] params, String[] signature) throws ReflectionException, MBeanException
+    {
+        return delegate().instantiate(className, params, signature);
+    }
+
+    @Override
+    public Object instantiate(String className, ObjectName loaderName, Object[] params, String[] signature) throws ReflectionException, MBeanException, InstanceNotFoundException
+    {
+        return delegate().instantiate(className, loaderName, params, signature);
+    }
+
+    @Override
+    public ObjectInputStream deserialize(ObjectName name, byte[] data) throws InstanceNotFoundException, OperationsException
+    {
+        return delegate().deserialize(name, data);
```

Comment:
Seems those `deserialize` methods are deprecated in favor of `getClassLoader`, did you consider that?

Reply:
this class is meant to proxy all requests to the base to simplify overloading classes, so shouldn't call a different method than expected.